pom.xml

```diff
@@ -1816,6 +1775,104 @@ This will force full Scala code rebuild in downstream modules.
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>get-buildvers</id>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <target>
+                                <script language="jython">
+import ConfigParser
+config = ConfigParser.ConfigParser()
+config.read("release.properties")
+<!-- #if scala-2.12 -->
+print(config.get("Scala212ReleaseSection", "${expression}"))
+<!-- #endif scala-2.12 -->
+<!-- #if scala-2.13 --><!--
+print(config.get("Scala213ReleaseSection", "${expression}"))
+--><!-- #endif scala-2.13 -->
+                                </script>
+                            </target>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>create-release-properties</id>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <phase>initialize</phase>
+                        <configuration>
+                            <target>
+                                <script language="jython">
+import subprocess
+import os
+<!-- #if scala-2.12 -->
+pom = ""
+<!-- #endif scala-2.12 -->
+<!-- #if scala-2.13 --><!--
+pom = "-f scala2.13"
+--><!-- #endif scala-2.13 -->
+
+# Define the Bash command
+bash_command = """
+export TEMP=$(mvn help:all-profiles -pl . {pom}| sort | uniq | awk "/release[0-9]/ {{print substr(\$3, 8)}}");
+TEMP=$(echo -n $TEMP)
+[[ -n $TEMP ]] || {{ echo -e "Error setting databricks versions"; }};
+&lt;&lt;&lt; $TEMP read -r -a SPARK_SHIM_VERSIONS_ARR;
+SNAPSHOTS=();
+NO_SNAPSHOTS=()
+for ver in ${{SPARK_SHIM_VERSIONS_ARR[@]}}; do
+    TEST=$(mvn -B help:evaluate -q -pl dist {pom} -Dexpression=spark.version -Dbuildver="$ver" -DforceStdout)
+    if [[ "$TEST" != *"-SNAPSHOT" ]]; then
+        NO_SNAPSHOTS+=(" $ver")
+    else
+        SNAPSHOTS+=(" $ver")
+    fi
```

Comment:
We can handle this and future cases where we don't need to include a specific profile in the release by just changing the profile name to something other than `release\d+`. To exclude 400 for example we can just change the profile to `pre-release400` which will then not be included. I have ran this example with and without including 400 

We aren't ready so we renamed the `release400` to `pre-release400` in pom.xml, aggregator/pom.xml and sql-plugin/pom.xml. 
```
mvn -B -q -pl . -f scala2.13 -Dexpression=snapshots.buildvers antrun:run@get-buildvers -DforceStdout


```
When we are ready to add it to the snapshots releases we change the profile back to `release400` 
```
mvn -B -q -pl . -f scala2.13 -Dexpression=snapshots.buildvers antrun:run@get-buildvers -DforceStdout
400
```

Does this work for you?

Reply:
Did you test this?  I pulled the branch and it doesn't work.  I modified release400 to pre-release400 in the scala2.13/pom.xml, but pre-release400 still matches the awk pattern /release[0-9]/ and the substr ends up mangling the string improperly because of the magic "8" that is wrong for that string.

Reply:
Yes, I tested it. I forgot to mention that the regex in the awk command needs to be modified to `awk '/([^-])release[0-9]/ {print substr($3, 8)}');` I will be pushing my changes soon, you can test on that

Reply:
Let's make that less hacky and change the string to check for what Maven actually emits, e.g.:
`/  Profile Id: release[0-9]/`, then we don't have top remember to use hyphens or other magic patterns.

Reply:
I would keep the profile names consistent, alone for the reason of avoiding regex bugs and easier grepping. And manage exclusion list: release.excluded, snapshot.excluded, etc. I expect them to have a couple of values at most